tendermint/src/lite.rs

```diff
@@ -3,3 +3,122 @@ pub mod verifier;
+        "evidence_hash": "",
+        "proposer_address": "30B3C78C25BB72796960B63AA06CA7ED5F648951"
+      }"#,
+        )
+        .unwrap();
+        let signed_header: SignedHeader = serde_json::from_str(r#"{
+      "header": {
+        "version": {
+          "block": "10",
+          "app": "0"
+        },
+        "chain_id": "test-chain-y3m1e6-AB",
+        "height": "1",
+        "time": "2019-11-19T07:56:45.46738Z",
+        "num_txs": "0",
+        "total_txs": "0",
+        "last_block_id": {
+          "hash": "",
+          "parts": {
+            "total": "0",
+            "hash": ""
+          }
+        },
+        "last_commit_hash": "",
+        "data_hash": "",
+        "validators_hash": "1F0F3E080DCE3253C88394F2CE479C074C5A25FE94A6C8560C76B6F2281BD685",
+        "next_validators_hash": "1F0F3E080DCE3253C88394F2CE479C074C5A25FE94A6C8560C76B6F2281BD685",
+        "consensus_hash": "048091BC7DDC283F77BFBF91D73C44DA58C3DF8A9CBC867405D8B7F3DAADA22F",
+        "app_hash": "2E419E5A3CDD1F718EA956E9AF91E57FD4FD5D8C8DDEA43FB3D8728E8E41852D",
+        "last_results_hash": "",
+        "evidence_hash": "",
+        "proposer_address": "30B3C78C25BB72796960B63AA06CA7ED5F648951"
+      },
+      "commit": {
+        "block_id": {
+          "hash": "0D06FAEB490248DC504CDD2B4318BEFCB2FD06F866BACAE9864CC5B7373D2A82",
+          "parts": {
+            "total": "1",
+            "hash": "C74C19586564ECC8A348BAE68D505230618B9FABB291549FFD59A6322E961865"
+          }
+        },
+        "precommits": [
+          {
+            "type": 2,
+            "height": "1",
+            "round": "0",
+            "block_id": {
+              "hash": "0D06FAEB490248DC504CDD2B4318BEFCB2FD06F866BACAE9864CC5B7373D2A82",
+              "parts": {
+                "total": "1",
+                "hash": "C74C19586564ECC8A348BAE68D505230618B9FABB291549FFD59A6322E961865"
+              }
+            },
+            "timestamp": "2019-11-19T07:57:02.408782Z",
+            "validator_address": "30B3C78C25BB72796960B63AA06CA7ED5F648951",
+            "validator_index": "0",
+            "signature": "C/AVaOIcMyXozkyMiWFAB2/hHZeBKCZ97W+ivbJtBlgwYxyVRZeQQhBF+8uxr0vvpbKDEKUIxaq8Zng1nN5LAA=="
+          }
+        ]
+      }
+    }"#).unwrap();
+        let last_validators: Vec<validator::Info> = serde_json::from_str(
+            r#"[
+        {
+          "address": "30B3C78C25BB72796960B63AA06CA7ED5F648951",
+          "pub_key": {
+            "type": "tendermint/PubKeyEd25519",
+            "value": "ZncnqXOpTbsTYZqkbdzHne9y1h8Z1gp3Xlayfi4QzW0="
+          },
+          "voting_power": "12500000000"
+        }
+      ]"#,
+        )
+        .unwrap();
+        let validators = last_validators.clone();
+        verify_trusting(
+            header,
+            signed_header,
+            validator::Set::new(last_validators),
+            validator::Set::new(validators),
+        )
+        .expect("verify_trusting failed");
+    }
```

Comment:
Very cool! Thanks. I'm wondering if we either should have simpler rust-only tests first. Or, simplify this (extract all json strings to vars/constants and have only the verification logic in the test). Also, there should be a comment explaining how the JSON was generated. 

Furthermore, @Shivani912 is working on generating similar tests: https://github.com/Shivani912/tendermint/tree/shivani/language-agnostic-tests/types/lite-client/tests

Reply:
I simply copy-pasted my local test node's json rpc response, only to test the basics work.
It's great to have more sophisticated test cases.

Reply:
I think, these kind of tests currently tend to live in https://github.com/interchainio/tendermint-rs/tree/master/tendermint/tests as JSON files: https://github.com/interchainio/tendermint-rs/tree/master/tendermint/tests/support/rpc

I think it would increase the readability here if we'd extract the JSON into vars in the test (e.g. ` static VALID_HEADER: &str =  r#"{ ...}"#`

The test seems to fail on my machine btw.